File: src/main/java/dev/slne/discord/exception/DiscordException.java
package dev.slne.discord.exception;

import java.io.Serial;

public abstract class DiscordException extends Exception {

  @Serial
  private static final long serialVersionUID = 7077818795975848866L;

  public DiscordException(String message) {
    super(message);
  }

  public DiscordException(String message, Throwable cause) {
    super(message, cause);
  }

  public DiscordException(DiscordException exception) {
    super(exception.getMessage(), exception);
  }
}

File: src/main/kotlin/dev/slne/discord/exception/DiscordException.kt
package dev.slne.discord.exception

import java.io.Serial

abstract class DiscordException : Exception {
    
    constructor(message: String?) : super(message)

    constructor(message: String?, cause: Throwable?) : super(message, cause)

    constructor(exception: DiscordException) : super(exception.message, exception)

    companion object {
        @Serial
        private const val serialVersionUID = 7077818795975848866L
    }
}
